app/controllers/api/v1/preferences_controller.rb

```diff
@@ -0,0 +1,32 @@
+module Api
+  module V1
+    # RESTful JSON API for preference actions
+    class PreferencesController < ApiController
+      before_action :authenticate_user!
+
+      def update
+        response = rest_preference_service.update(preference_params.merge(id: params[:id]))
+        if response
+          render json: true
+        else
+          render status: 422, json: false
+        end
+      end
+
+      private
+
+      def preference_params
+        params.require(:preference).permit(
+          :id,
+          :individual_preference,
+          :post_lottery_validation,
+          :type_of_proof,
+        )
+      end
```

Comment:
Similar to my comment above, here we only allow the set of params that we need in the Salesforce standard REST API preference update requests that we have so far. So far, we only let users update three fields on preferences from the Confirmed Preference panels.